src/dex/infusion/infusion.ts

```diff
@@ -0,0 +1,695 @@
+import { isStablePair } from './utils/isStablePair';
+
+export enum InfusionRouterFunctions {
+  sellExactEth = 'swapExactETHForTokens',
+  sellExactToken = 'swapExactTokensForETH',
+  swapExactIn = 'swapExactTokensForTokens',
+}
+
+export interface SmardexData extends Omit<UniswapV2Data, 'feeFactor'> {
+  deadline: number;
+  receiver: Address;
+}
+
+const VelodromeFactoryABI = [
+  {
+    inputs: [{ internalType: 'bool', name: '_stable', type: 'bool' }],
+    name: 'getFee',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+const velodromeFactoryIface = new Interface(VelodromeFactoryABI);
+const erc20Iface = new Interface(erc20ABI);
+const infusionPairIface = new Interface(infusionPairABI);
+const defaultAbiCoder = new AbiCoder();
+
+export class Infusion extends UniswapV2 {
+  pairs: { [key: string]: InfusionPair } = {};
+  stableFee?: number;
+  volatileFee?: number;
+
+  readonly isFeeOnTransferSupported: boolean = true;
+  readonly SRC_TOKEN_DEX_TRANSFERS = 1;
+  readonly DEST_TOKEN_DEX_TRANSFERS = 1;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(_.pick(InfusionConfig, ['Infusion']));
+
+  constructor(
+    protected network: Network,
+    dexKey: string,
+    protected dexHelper: IDexHelper,
+    isDynamicFees = true,
+    factoryAddress?: Address,
+    subgraphURL?: string,
+    initCode?: string,
+    feeCode?: number,
+    poolGasCost?: number,
+    routerAddress?: Address,
+  ) {
+    super(
+      network,
+      dexKey,
+      dexHelper,
+      isDynamicFees,
+      factoryAddress !== undefined
+        ? factoryAddress
+        : InfusionConfig[dexKey][network].factoryAddress,
+      subgraphURL === ''
+        ? undefined
+        : subgraphURL !== undefined
+        ? subgraphURL
+        : InfusionConfig[dexKey][network].subgraphURL,
+      initCode !== undefined
+        ? initCode
+        : InfusionConfig[dexKey][network].initCode,
+      feeCode !== undefined ? feeCode : InfusionConfig[dexKey][network].feeCode,
+      poolGasCost !== undefined
+        ? poolGasCost
+        : InfusionConfig[dexKey][network].poolGasCost,
+      infusionPairIface,
+      Adapters[network] || undefined,
+    );
+
+    this.stableFee = InfusionConfig[dexKey][network].stableFee;
+    this.volatileFee = InfusionConfig[dexKey][network].volatileFee;
+
+    this.factory = new dexHelper.web3Provider.eth.Contract(
+      infusionFactoryABI as any,
+      factoryAddress !== undefined
+        ? factoryAddress
+        : InfusionConfig[dexKey][network].factoryAddress,
+    );
+
+    this.router =
+      routerAddress !== undefined
+        ? routerAddress
+        : InfusionConfig[dexKey][network].router || '';
+
+    this.feeFactor =
+      InfusionConfig[dexKey][network].feeFactor || this.feeFactor;
+
+    // this.exchangeRouterInterface = new Interface(infusionRouterABI);
```

Comment:
@JackieJoo Let's remove this commented line if it's not needed

Reply:
done